src/sentry/api/serializers/models/releaseactivity.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+from typing import Any, Mapping, MutableMapping, Sequence, TypedDict
+
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models import Group
+from sentry.models.releaseactivity import ReleaseActivity
+from sentry.utils.json import JSONData
+
+
+class ReleaseActivityResponse(TypedDict):
+    id: int
+    type: str
+    data: Mapping[str, JSONData]
+    date_added: datetime
+
+
+@register(ReleaseActivity)
+class ReleaseActivitySerializer(Serializer):
+    def get_attrs(
+        self, item_list: Sequence[ReleaseActivity], user: Any, **kwargs: Any
+    ) -> MutableMapping[Any, Any]:
+        def _update_data(d: MutableMapping[str, Any]):
+            if d.get("group_id"):
+                try:
+                    d["group"] = Group.objects.get(id=d.get("group_id"))
```

Comment:
This is really ugly .. 
Instead of serializing the group and storing it in the db, we only store the `group_id` and when we return the response, we'll retrieve the group and serialize it there. 

Not sure if there's a better way.

If we serialize the whole group and persist it in the DB, its essentially snapshoting the group details at the time when it gets stored in the DB vs only storing the `group_id` and retrieving the details with the freshest state.